packages/types/src/config.ts

```diff
@@ -45,3 +45,12 @@ export interface SettingsOptions {
   testingType?: 'component' |'e2e'
   args?: AllModeOptions
 }
+
+export type BannerState = {
+  lastShown?: number | null
```

Comment:
Should we just drop the `?`? Or, make the type `| null` - otherwise `undefined` and `null` are both okay, maybe it's good to have one way to show false-y-ness.

Could be more to it, not a big deal, just call this out whenever I see it.

Reply:
Hmm, I'll give this some more thought. My original thought was to mimic the GQL way of "anything can be undefined or null by default", but since we aren't directly querying this in GQL (just the encapsulating JSON object) I guess that isn't set in stone

Reply:
Right, I see... I guess that makes sense, `null` means "It's not defined" and `undefined` means "it's not here since you didn't request it" - similar to JSON.

That said, not sure we want GraphQL leaking into places like `packages/types`, wonder if it makes sense to do `lastShown?: number`, then in the GraphQL resolver, coerce using `?? null` (since the `null` is a GraphQL concern).

Not a big deal, I really like to keep concerns nice and compartmentalized but since we can't realistically lint for this it's not exactly practical to do, even if we did agree on this pattern.

Reply:
You've convinced me 😁 . Done in [`da3c2e3`](https://github.com/cypress-io/cypress/pull/23256/commits/da3c2e3be43d84ccb4816bc2a08b341c394462d3)